### PR TITLE
Dh 10

### PR DESCRIPTION
**Pull request by Dyrk84**

### CODE REVIEW

File: src/main/java/hu/dungeonhunter/characters/Champion.java [hunk 1]
```diff
@@ -1,52 +1,50 @@
...
+    public void enemyVictory(){
+        if (hp <= 0){
+            System.out.println("You are soooo dead! Game Over!");
+            defeat = true;
+        }
     }
 
-    public int getChampionMaxDamage() {
-        return championMaxDamage;
+    void startValues() {
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/characters/Champion.java]: Keep methods private if not accessed from other classes.

**Comment by gergelyszanto** [comment]
Two of your tests are failing:
image

File: src/main/java/hu/dungeonhunter/tools/Fight.java [hunk 2]
```diff
@@ -1,47 +1,100 @@
...
-        System.out.print("Champion attack: ");
-        monster.setMonsterHP(monster.getMonsterHP() - champion.championDamage());
-        System.out.println("Monster have now " + monster.getMonsterHP() + " hit points");
-        if (monster.getMonsterHP() > 0)
-            monsterAttack();
-        else {
-            gameOver();
+    public Fight(){
+        monsterCounter = Dice.rollDice(6,2);
+        System.out.println(monsterCounter + " monsters is in the Dungeon!");
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/tools/Fight.java]:
```suggestion
        System.out.println(monsterCounter + " monsters are in the Dungeon!");
```

File: src/main/java/hu/dungeonhunter/tools/Fight.java [hunk 3]
```diff
@@ -1,47 +1,100 @@
...
+            actionMenu();
         }
+        else
+        System.out.println("The Dungeon is clear! You killed " + killedMonsterCounter + "monster (not counting the " +
+                "many mothers and children), you win!");
+    }
+
+    public Monsters monsterCaller() {
+        if (monsterCounter != 0) {
+            System.out.println("A monster step out from darkness!");
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/tools/Fight.java]:
```suggestion
            System.out.println("A monster steps out from darkness!");
```

File: src/main/java/hu/dungeonhunter/tools/Fight.java [hunk 4]
```diff
@@ -1,47 +1,100 @@
...
+            monster = new Monsters();
+        }return monster;
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/tools/Fight.java]:
```suggestion
        }
        return monster;
```

File: src/main/java/hu/dungeonhunter/tools/Fight.java [hunk 5]
```diff
@@ -1,47 +1,100 @@
...
-            monsterAttack();
-        else {
-            gameOver();
+    public Fight(){
+        monsterCounter = Dice.rollDice(6,2);
+        System.out.println(monsterCounter + " monsters is in the Dungeon!");
+        start();
+    }
+
+    public void start(){
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/tools/Fight.java]: Let's rename this to spawnMonster or something like that. start is not a verbose name (looking at the method name you don't know what it does start)

File: src/main/java/hu/dungeonhunter/tools/Fight.java [hunk 6]
```diff
@@ -1,47 +1,100 @@
...
+    int chosenNumber() {
+        do {
+            Scanner scanner = new Scanner(System.in);
+            try {
+                return scanner.nextInt();
+            } catch (InputMismatchException hibafogo) {
+                yourChooseIsNotAppropriate();
+                printMenu();
+            }
+        } while (true);
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/tools/Fight.java]:
```suggestion
        do {
            printMenu();
            Scanner scanner = new Scanner(System.in);
            try {
                return scanner.nextInt();
            } catch (InputMismatchException hibafogo) {
                yourChooseIsNotAppropriate();
            }
        } while (true);
```
And remove printMenu(); from line 93

**Comment by gergelyszanto** [comment]
Add test to validate monster counter (monster defeat)

File: src/main/java/hu/dungeonhunter/characters/Champion.java [hunk 7]
```diff
@@ -1,52 +1,50 @@
...
+    private boolean defeat;
 
-    public int getChampionHP() {
-        return championHP;
+    public Champion() {
+        this.hp = START_HP;
+        this.maxDamage = 6;
+        this.numOfDices = 1;
+        startValues();
+        this.defeat = false;
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/characters/Champion.java]: No need to give false initializer value to a boolean. When you say `private boolean defeat;` at line 23 it already has false value.

File: src/main/java/hu/dungeonhunter/characters/Monsters.java [hunk 8]
```diff
@@ -1,51 +1,47 @@
...
+    private boolean defeat;
 
-    public int getMonsterHP() {
-        return monsterHP;
+    public Monsters() {
+        this.hp = Dice.rollDice(4,3);
+        this.maxDamage = 6;
+        this.numOfDices = 1;
+        startValues();
+        this.defeat = false;
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/characters/Monsters.java]: Same here as at champion, no need for false initializer value.

File: src/main/java/hu/dungeonhunter/tools/Fight.java (hunk 2 above)
Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/tools/Fight.java]: Thank you, i corrected!

File: src/main/java/hu/dungeonhunter/tools/Fight.java (hunk 3 above)
Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/tools/Fight.java]: Thank you, i corrected this too!

File: src/main/java/hu/dungeonhunter/tools/Fight.java (hunk 4 above)
Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/tools/Fight.java]: Does this need to make the code look better?
I corrected.

Comment (gergely-szanto-epam) [reply src/main/java/hu/dungeonhunter/tools/Fight.java]: Basically yes. Easier to see the if block and the return value if separated.

File: src/main/java/hu/dungeonhunter/tools/Fight.java (hunk 6 above)
Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/tools/Fight.java]: Why? For dont take 2 line the printMenu method ?
I corrected.

Comment (gergely-szanto-epam) [reply src/main/java/hu/dungeonhunter/tools/Fight.java]: Yes, it is enough to call printMenu at one place. When the user is asked to enter an input, the menu will be automatically printed for the user.

**Pull request closed without merging**